File: app/src/release/java/com/app/missednotificationsreminder/CustomApplication.java
package com.app.missednotificationsreminder;

/**
 * Custom application implementation to provide additional functionality such as dependency injection, leak inspection
 *
 * @author Eugene Popovich
 */
public class CustomApplication extends CustomApplicationBase {

    @Override Object[] getModules() {
        return Modules.list(this);
    }
}

File: app/src/main/java/com/app/missednotificationsreminder/CustomApplication.kt
package com.app.missednotificationsreminder

import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

/**
 * Custom application implementation to provide additional functionality such as dependency injection, leak inspection
 *
 * @author Eugene Popovich
 */
class CustomApplication : CustomApplicationBase() {
    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerApplicationComponent.factory().create(applicationContext)
    }
}